com.unity.ml-agents/Runtime/Sensors/StackingSensor.cs

```diff
@@ -67,11 +63,40 @@ public StackingSensor(ISensor wrapped, int numStackedObservations)
             }
 
             // TODO support arbitrary stacking dimension
-            m_Shape[0] *= numStackedObservations;
-            m_StackedObservations = new float[numStackedObservations][];
-            for (var i = 0; i < numStackedObservations; i++)
+            m_Shape[m_Shape.Length - 1] *= numStackedObservations;
+
+            if (m_WrappedSensor.GetCompressionType() == SensorCompressionType.None)
             {
-                m_StackedObservations[i] = new float[m_UnstackedObservationSize];
+                m_StackedObservations = new float[numStackedObservations][];
+                for (var i = 0; i < numStackedObservations; i++)
+                {
+                    m_StackedObservations[i] = new float[m_UnstackedObservationSize];
+                }
+            }
+            else
+            {
+                m_StackedCompressedObservations = new byte[numStackedObservations][];
+                m_EmptyCompressedObservation = CreateEmptyPNG();
+                for (var i = 0; i < numStackedObservations; i++)
+                {
+                    m_StackedCompressedObservations[i] = m_EmptyCompressedObservation;
+                }
+
+                // Construct compression mapping
+                var compressibleSensor = m_WrappedSensor as ICompressibleSensor;
+                var wrappedMapping = compressibleSensor.GetCompressionMapping();
+                int wrappedNumChannel = shape[2];
+                int wrappedMapLength = wrappedMapping.Length;
+                int offset;
+                m_CompressionMapping = new int[wrappedMapLength * m_NumStackedObservations];
+                for (var i = 0; i < numStackedObservations; i++)
```

Comment:
You need to add some C# unit tests to make sure the mapping results in what you expect.